resources/ceph_block_pool.py

```diff
@@ -0,0 +1,51 @@
+"""
+General Ceph block pool class
+"""
+import yaml
+import logging
+from resources.base_resource import BaseOCSClass
+from ocs import defaults
+from utility.templating import load_yaml_to_dict
+
+
+log = logging.getLogger(__name__)
+
+
+class CephBlockPool(BaseOCSClass):
+    """
+    CephBlockPool kind class
+    Provides basic methods to operate on CephBlockPool kind.
+    """
+    def __init__(self, **kwargs):
+        """
+        Initializer method
+
+        kwargs:
+            Copy of ocs/defaults.py::CEPHBLOCKPOOL_DICT dictionary
+        """
+        self.cbp_data = defaults.CEPHBLOCKPOOL_DICT
+        self.cbp_data.update(kwargs)
+        super(CephBlockPool, self).__init__(
+            self.cbp_data.get('apiVersion'), self.cbp_data.get('kind'),
+            self.cbp_data.get('metadata').get('namespace')
+        )
+        self._name = self.cbp_data.get('metadata').get('name')
+
+    @property
+    def name(self):
+        return self._name
+
+    def reload(self):
```

Comment:
Would it make sense to move methods `reload()`, `get()` and `delete()` (and probably `create()`) to `BaseOCSClass` so it can be accessed by all children of the class?

Reply:
not all of them but probably delete and get can be moved.
Create - every class will implement differently the creation of the resource based on several things like the ceph interface, etc
reload - probably can move as well but as there is no actual usage of it, yey, we can't tell

Reply:
@fbalak Yes, I agree, will do that

Reply:
@dahorak @RazTamir for now everything is generic so moved all the base class, see next commit

Reply:
@RazTamir I think that in the case of storageclass, we can do the distinction between block and file outside the object creation method. For example, in the helpers function of storageclass creation, we'll send a copy of defaults.STORAGECLASSCEFFS_DICT which will contain the relevant keys for storageclss of ceph fs interface type

Reply:
So why do we need the kind classes if we are going to use functionality from helpers?

Reply:
@RazTamir for future implementation for methods which will be relevant for specific kinds. Not sure if there will be any though

Reply:
I'm not sure this is a good direction as we wanted to simplify the usage of kinds, with that change we just wrapped the basic OCP methods to enable also yaml dump/load - that's it.